pupil_src/shared_modules/video_capture/file_backend.py

```diff
@@ -372,6 +444,21 @@ def recent_events_own_timing(self, events):
     @ensure_initialisation()
     def seek_to_frame(self, seek_pos):
         # frame accurate seeking
+        ori_pos = seek_pos
+        container_index = 0
+        # self.frame_count contain frame_count from every video
+        # like [50, 100, 150], so when we get a seek_pos 120, we
+        # calculate which video we should play
+
+        for i in range(len(self.frame_count)-1, 0, -1):
+            if seek_pos >= sum(self.frame_count[:i]):
```

Comment:
This seems a bit too complicated to find the correct index. I suggest using `np.cumsum` + `bisect`/`np.searchsorted` to find the correct container index.